src/Workspaces/SharedUtilitiesAndExtensions/Workspace/Core/Extensions/TextDocumentExtensions.cs

```diff
@@ -2,13 +2,25 @@
 // The .NET Foundation licenses this file to you under the MIT license.
 // See the LICENSE file in the project root for more information.
 
+using System.Threading.Tasks;
+using System.Threading;
 using Microsoft.CodeAnalysis.Text;
 using Roslyn.Utilities;
 
 namespace Microsoft.CodeAnalysis.Shared.Extensions
 {
     internal static partial class TextDocumentExtensions
     {
+#if CODE_STYLE
+        public static ValueTask<SourceText> GetValueTextAsync(this TextDocument document, CancellationToken cancellationToken)
+        {
+            if (document.TryGetText(out var text))
+                return ValueTaskFactory.FromResult(text);
+
+            return new ValueTask<SourceText>(document.GetTextAsync(cancellationToken));
+        }
+#endif
```

Comment:
i would be supportive of you makign an API proposal for this and going through public API review as well.  (not blocking this PR of course).